pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -165,31 +165,41 @@ func (ks *scaler) handleScaleToZero(ctx context.Context, pa *pav1alpha1.PodAutos
 
 	now := time.Now()
 	logger := logging.FromContext(ctx)
-	if pa.Status.IsActivating() { // Active=Unknown
+	switch {
+	case pa.Status.IsActivating(): // Active=Unknown
 		// If we are stuck activating for longer than our progress deadline, presume we cannot succeed and scale to 0.
 		if pa.Status.CanFailActivation(now, activationTimeout) {
 			logger.Info("Activation has timed out after ", activationTimeout)
 			return desiredScale, true
 		}
 		ks.enqueueCB(pa, activationTimeout)
 		return scaleUnknown, false
-	} else if pa.Status.IsReady() { // Active=True
+	case pa.Status.IsReady(): // Active=True
 		// Don't scale-to-zero if the PA is active
 		// but return `(0, false)` to mark PA inactive, instead.
 		sw := aresources.StableWindow(pa, cfgAS)
 		af := pa.Status.ActiveFor(now)
 		if af >= sw {
-			// We do not need to enqueue PA here, since this will
-			// make SKS reconcile and when it's done, PA will be reconciled again.
+			// If SKS currently is in Serving mode, we do not need to enqueue PA here,
+			// since SKS will reconcile to change the mode and when it's done,
+			// PA will be reconciled again.
+			// Othwerwise, SKS might not meaningfully change and thus
+			// PA will not be re-enqueued in time.
+			// So enqueue PA for reconcile again in a few seconds.
+			if sks.Spec.Mode == nv1a1.SKSOperationModeProxy {
```

Comment:
The comment says the reenqueue condition is that the SKS it not in serving mode. Can we keep consistent for this if condition and the comments?

Reply:
Well it's the same. We have only 2 modes, but I can use same wording

Reply:
I prefer to use the same wording in case we have more than 2 modes in the future.

Reply:
Done.